jobs/build.Jenkinsfile

```diff
@@ -231,8 +231,28 @@ lock(resource: "build-${params.STREAM}") {
         // fetch from repos for the current build
         stage('Fetch') {
             shwrap("""
-            cosa fetch ${strict_build_param}
-            """)
+            # Dont run this for production builds
+            if (stream_info.type != "production") {
+                # Last author time in UTC
+                echo "Last commit: $(date -ud @$(git log -1 --follow manifest-lock* --format="%ct"))" 
+                # Number of minutes to check
+                n=120
+                # Current time
+                current_time=$(date +%s)
+                # Time n minutes ago
+                n_minutes_ago=$(( current_time - ( 60 * n ) ))
+                # Last commit
+                last_git_log=$(git log -1 --follow manifest-lock* --format="%cI")
+                # Check if git log was run within the last n minutes
+                if (( last_git_log > n_minutes_ago )); then
```

Comment:
let's export this logic into another function and just call it.. the code here would then look something like:

```
def overrides_fetch_param = ""
if (stream_info.type != "production" && recent_commits_to_lockfiles()) {
    overrides_fetch_param = "--with-cosa-overrides"
}
shwrap("""
cosa fetch ${overrides_fetch_param} ${strict_build_param}
""")
```